makelib/golang.mk

```diff
@@ -241,21 +263,36 @@ go.vendor: $(DEP)
 	@$(DEP) ensure || $(FAIL)
 	@$(OK) dep ensure
 
+go.mod.verify:
+	@$(INFO) verify dependencies have expected content
+	@$(GO) mod verify || $(FAIL)
+	@$(OK) go modules dependencies verified
+
+go.mod.update:
+	@$(INFO) update go modules
+	@$(GO) get -u ./... || $(FAIL)
+	@$(OK) update go modules
+
+go.mod.vendor:
+	@$(INFO) go mod vendor
+	@$(GO) mod vendor || $(FAIL)
+	@$(OK) go mod vendor
+
 go.clean:
+	@# make all read-only go modules folder writable, so they can be deleted
```

Comment:
I don't follow why this needed if everything goes in the WORK_DIR.

Reply:
if you have `dir1/some/path/dir2/stuff`, where `dir2` is readonly, `rm -rf dir1` will return `Permission denied`

Reply:
This later causes the `make clean` command fail in Jenkins

Reply:
none of the paths in WORK_DIR should be read only. Do you know which path is failing?

Reply:
The `GOPATH/pkg/mod` is set to readonly by go, when installing modules
https://github.com/golang/go/issues/27455

Reply:
Also https://github.com/golang/go/issues/27161

I tried using `go clean -modcache` suggested [here](https://github.com/golang/go/issues/27161#issuecomment-415213240), but it wont' work since `make clean` is run from the Jenkins environment, which invokes `go` installed in Jenkins (which is outdated)

Reply:
this is the example of Jenkin's complaining
https://jenkinsci.upbound.io/blue/organizations/jenkins/crossplane%2Fstack-aws/detail/PR-20/66/pipeline

Reply:
It's the directories which are causing issues, whereas `chmod -R ...` recursively updates both files and folders. If the number of files are large (all the files in downloaded go packages), this could take some time. Do you still rather do that?